index.js

```diff
@@ -1,6 +1,103 @@
 /* jshint node: true */
-'use strict';
+
+var _defaultConfiguration = {
+  googleAnalytics: {
+    globalVariable: 'ga',
+    webPropertyId: null,
+    cookieDomain: null,
+    cookieName: null,
+    cookieExpires: null,
+    displayFeatures: null,
+    linkid: null
+  }
+};
+
+function mergeConfiguration(defaultConfiguration, config) {
+  var key;
+
+  defaultConfiguration || (defaultConfiguration = {});
+  config || (config = {});
+
+  for (key in defaultConfiguration) {
+    if (!defaultConfiguration.hasOwnProperty(key)) {
+      continue;
+    }
+
+    if (!config[key]) {
+      config[key] = defaultConfiguration[key];
+    }
+  }
+
+  return config
+}
+
+function getGoogleAnalyticsTrackingCode(config) {
+  var scriptArray, key, value, linkid = false, displayFeatures = false, gaConfig = {};
+
+  for (key in config) {
+    if (!config.hasOwnProperty(key)) {
+      continue;
+    }
+
+    value = config[key];
+
+    if (value === null) {
+      continue;
+    }
+
+    if (key === 'linkid') {
+      linkid = value;
+      delete config[key];
+    } else if (key === 'displayFeatures') {
+      displayFeatures = value;
+      delete config[key];
+    }
+
+    gaConfig[key] = config[key];
+  }
+
+  if (Object.keys(gaConfig).length === 0) {
+    gaConfig = "'auto'";
+  } else {
+    gaConfig = JSON.stringify(gaConfig);
+  }
+
+  scriptArray = [
+    "<script>",
+    "(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){",
+    "(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),",
+    "m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)",
+    "})(window,document,'script','//www.google-analytics.com/analytics.js','" + config.globalVariable + "');",
+    "",
+    displayFeatures ? config.globalVariable + "('require', 'displayfeatures');" : "",
+    linkid ? config.globalVariable + "('require', 'linkid', 'linkid.js');" : "",
+    "" + config.globalVariable + "('create', '" + config.webPropertyId + "', " + gaConfig + ");",
+    "</script>"
+  ];
+
+  return scriptArray;
+}
 
 module.exports = {
-  name: 'ember-insights'
+  name: 'ember-insights',
+
+  contentFor: function (type, config) {
+    var content = [], trackers, gaConfig;
+
+    config.analyticsInsights || (config.analyticsInsights = {});
+    trackers = config.analyticsInsights.trackers || (config.analyticsInsights.trackers = {});
+
+    if (trackers.googleAnalytics) {
+      gaConfig = mergeConfiguration(
+        _defaultConfiguration.googleAnalytics,
+        trackers.googleAnalytics || {}
+      );
+
+      if (type === 'head' && gaConfig.webPropertyId !== null) {
```

Comment:
@RuslanZavacky, it is not always possible to create GA tracker in index.html. Sometimes information from target route is required for tracker creation. Take a look at this example:
```coffeescript
Insights.configure(
  # ...
).track(
  name: 'pageWithCustomWebProperty'
  insights:
    TRANSITIONS: ['page_with_custom_property']
  handler: (type, data, tracker) ->
    page = data.route.get('controller.model')
    gaFunc = tracker.getTracker()

    # Creation of tracker !!!
    gaFunc('create', page.get('ga_web_property'))

    tracker.trackPageView "/#{page.get('id')}"
)
```

But `analytics.js` script itself can be loaded in `index.html` even if `webPropertyId` is not available yet.

Can you please change your code so that it loads `analytics.js` regardless of whether webPropertyId is available in gaConfig.